### PR TITLE
[AutoML] Sweep Trimming of Whitespace in AutoML

### PR DESCRIPTION
**Pull request by justinormont**
Fixes https://github.com/dotnet/machinelearning/issues/3917:
> We should iterate over the options for TrimWhitespace in the TextLoader.
> 
> As diagnosed by @daholste, the MSLR-WEB10K dataset ([train](https://aka.ms/mlnet-resources/benchmarks/MSLRWeb10KTrain720kRows.tsv), [validate](https://aka.ms/mlnet-resources/benchmarks/MSLRWeb10KValidate240kRows.tsv), [test](https://aka.ms/mlnet-resources/benchmarks/MSLRWeb10KTest240kRows.tsv), [zip](https://express-tlcresources.azureedge.net/datasets/MSLR-WEB10K/MSLR-WEB10K.zip), [README](https://express-tlcresources.azureedge.net/datasets/MSLR-WEB10K/README.md), [LICENSE](https://express-tlcresources.azureedge.net/datasets/MSLR-WEB10K/LICENSE.md)) has an extra tab at the end of each data row, though not the header row. 
> 
> This causes the AutoML TextLoader to fail w/ the following error: 
> ```
> An Error occured during inferring columns
> Unable to split the file provided into multiple, consistent columns.
> Please see the log file for more info.
> Exiting ...
> ```
> 
> Work:
> * Have AutoML sweep over the [TrimWhitespace](https://github.com/dotnet/machinelearning/blob/429f8cc7764769fcf8c7f3668cc0a27619ec9531/src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs#L1449) option
> * Fix above error message -- spelling of "occured"

### CODE REVIEW

File: src/mlnet/Strings.resx [hunk 1]
```diff
@@ -193,10 +193,10 @@
     <value>Check out log file for more information</value>
   </data>
   <data name="ErrorGeneratingProject" xml:space="preserve">
-    <value>Exception occured while generating the project.</value>
+    <value>Exception occurred while generating the project.</value>
```
Comment (daholste) [inline src/mlnet/Strings.resx]: thx for fixing all of these

File: src/Microsoft.ML.AutoML/ColumnInference/TextFileContents.cs [hunk 2]
```diff
@@ -50,12 +50,14 @@ public static ColumnSplitResult TrySplitColumns(MLContext context, IMultiStreamS
         {
             var sparse = new[] { false, true };
             var quote = new[] { true, false };
+            var trim = new[] { false, true };
             var foundAny = false;
             var result = default(ColumnSplitResult);
             foreach (var perm in (from _allowSparse in sparse
                                     from _allowQuote in quote
+                                    from _trim in trim
```
Comment (daholste) [inline src/Microsoft.ML.AutoML/ColumnInference/TextFileContents.cs]: wow, thank you!!

**Comment by Dmitry-A** [comment]
All good code-wise but needs a test case or two.

File: src/mlnet/strings.Designer.cs [hunk 3]
```diff
@@ -10,48 +10,35 @@
...
 namespace Microsoft.ML.CLI {
     using System;
+    using System.Reflection;
     
     
-    /// <summary>
-    ///   A strongly-typed resource class, for looking up localized strings, etc.
-    /// </summary>
-    // This class was auto-generated by the StronglyTypedResourceBuilder
-    // class via a tool like ResGen or Visual Studio.
```
Comment (daholste) [inline src/mlnet/strings.Designer.cs]: weird about the differences in this file. if it looks like nothing of substance changed here, should we revert this file?

Comment (justinormont) [reply src/mlnet/strings.Designer.cs]: Agreed that it's odd. There are intended changes, the misspellings. But it seems that rebuilding also re-auto-generated the file, and now the form is a bit different.

Comment (justinormont) [reply src/mlnet/strings.Designer.cs]: @Dmitry-A -- any idea why this auto-generated file changed so much?

Comment (daholste) [reply src/mlnet/strings.Designer.cs]: Makes sense that intended changes caused a rebuild. I was noticing that (I think) the misspellings didn't directly change in this file? I was wondering if we reverted the file if the spelling bugs would still be fixed lol. I wonder if this is a Mac vs Windows Visual Studio difference? When I sync your branch on my Windows machine and rebuild the file, it looks like the original file

Comment (daholste) [reply src/mlnet/strings.Designer.cs]: ah I see, the typos were in the comments in this class. perhaps I can push the version I have built on my machine

**Comment by justinormont** [comment]
@harishsk: This could use a cherry pick into master, though should be tested w/ the CLI against various datasets to ensure they are still working properly.

**Comment by harishsk** [comment]
@justinormont This does look like it should be merged to master. Can you please create a PR after it has been tested with the CLI against the relevant databases?

**Pull request merged**